lib/cli/commands.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016 Resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+* @module Versionist.CLI.commands
+**/
+
+const landr = require('../landr');
+const helpers = require('./helpers');
+const Promise = require('bluebird');
+const ghpages = Promise.promisifyAll(require('gh-pages'));
+const fs = Promise.promisifyAll(require('fs-extra'));
+const chalk = require('chalk');
+
+/**
+* @summary Promise chain for compiling + serving website
+* @function
+* @public
+*
+* @param {Object} argv - arguments from the cli
+* @returns {Promise} - Empty resolve
+*
+* @throws If try catch breaks the chain.
+*
+* @example
+* lander.dev({ port: 3000, silent: true}).then(() => {
+*   console.log('Your website is served');
+* });
+*/
+
+exports.dev = (argv) => {
+  landr.getCompiler(argv)
+  .then((compiler) => {
+    return landr.serve(argv, compiler);
+  })
+  .then(() => {
+    console.log(chalk.green('Successfully compiled'));
+    console.log(chalk.green(`Successfully serving on port: ${argv.port}`));
+  })
+  .catch((err) => {
+    return helpers.showErrorAndQuit(err);
+  });
+};
+
+/**
+* @summary Promise chain for compiling + deploying website
+* @function
+* @public
+*
+* @param {Object} argv - arguments from the cli
+* @param {Object} compiler - webpack compiler
+* @returns {Promise} - Empty resolve
+*
+* @throws If try catch breaks the chain.
+*
+* @example
+* lander.deploy({ port: 3000, silent: true}, compiler).then(() => {
+*   console.log('Your website is deployed');
+* });
+*/
+
+exports.deploy = (argv) => {
+  // always compile for prod
+  argv.prod = true;
+
+  Promise.all([
+    fs.ensureDirAsync(argv.buildDir),
+
+    // remove cached files, gh-pages throws if other files are found
+    fs.removeAsync(`${__dirname}/../node_modules/gh-pages/.cache`)
+  ])
+  .then(() => {
+    return landr.getCompiler(argv);
+  })
+  .then((compiler) => {
+    // Manually create promise because return Promise.promisify(compiler.run)(); doesn't work :/
+    return new Promise(function(resolve, reject) {
```

Comment:
It'd be nicer if the promisify workaround wasn't here, so this code just said `compiler.run`, and was a little easier to quickly understand (and so we didn't need to ever think about this again if we need to use compiler.run elsewhere).

If you move this logic to exports.getCompiler instead, this gets much cleaner.